palindrome.c

```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
+#include <string.h>
+
+/*initial left = 0, initial right = strlen(w)-2*/
+int pal(char w[100], int left, int right){
+	if(strlen(w)/2==left || strlen(w)/2==right || right==left)
+		return 1;
+	if(w[right]!=w[left])
+		return 0;
+	left++;
+	right--;/*checks next letters*/
+	pal(w, left, right);
```

Comment:
the reason for your warning was that you should have returned the pal function call instead.